### ISSUE TITLE
Discuss interaction of replayability and interactivity

### ISSUE THREAD

**Issue by tfpauly** [ietf-wg-ohai/draft-ohai-chunked-ohttp#27]
https://github.com/ietf-wg-ohai/draft-ohai-chunked-ohttp/pull/25#pullrequestreview-2370940482

From @martinthomson 's comment:

> If a message can be replayed, then how that is replayed might be exploited by a malicious relay to effect change.
> 
> Maybe we need to do a little more analysis on that aspect of this. A section on interactivity and replay might help expound on those risks.

**Comment by tfpauly** [comment]
My understanding is that in the cases where there is no interaction, just a chunked response after a request completes, there isn't any new impact of a malicious replay by a relay (assuming the request is idempotent).

In the interactive case, agreed that more analysis would be good.

**Comment by martinthomson** [comment]
The core problem is that interactivity is really something that a client decides to do.  Though a server can observe the effect (we've covered that in the interactivity text we have now) that doesn't make interactive cases safe from replay.  Consider the case where there is a 100-continue in a request.  If that request is completed, it can still be replayed by the relay.  Its interactivity did not make it safer.

Note that a interactivity is one of the things that TLS depends on, so some people might assume that this design gains similar benefits to what TLS provides.  Because we don't build in a dependency on interactivity, we don't get the liveness guarantees that TLS does.

I think that's all there really is to say about that. I've spent a bit of time thinking about it, but I haven't come up with anything else.
